app/views/automated_tests/student_interface.html.erb

```diff
@@ -4,15 +4,15 @@
 <%= javascript_include_tag 'result', nonce: true %>
 <%= javascript_include_tag 'Results/main', nonce: true %>
 
-<% if @assignment.enable_student_tests && !@grouping.nil? %>
+<% if policy(@student).run_tests? %>
```

Comment:
You should revert this change; my previous comment about the `run_tests?` policy was referring to the code on line 15, not this line. (See my new comment below.)